TESTS/mbed_hal/spi/main.cpp

```diff
@@ -0,0 +1,560 @@
+                    /* Wait until the end of the spi transmission. */
+                }
+
+                TEST_ASSERT_EQUAL(true, handler_called);
+            }
+        }
+
+    }
+    spi_free(&spi_obj);
+}
+
+/* Test that spi_transfer_async_abort() can successfully abort an on-going async transfer. */
+void test_transfer_master_async_abort()
+{
+    spi_t spi_obj = {0};
+    spi_capabilities_t capabilities = {0};
+
+    const DMAUsage dma_modes[] = {
+        DMA_USAGE_NEVER,
+        DMA_USAGE_OPPORTUNISTIC,
+        DMA_USAGE_ALWAYS,
+        DMA_USAGE_TEMPORARY_ALLOCATED,
+        DMA_USAGE_ALLOCATED
+    };
+
+    void *p_tx_buf;
+    void *p_rx_buf;
+    void *p_fill_sym;
+
+    spi_get_capabilities(spi_get_module(SPI_MASTER_MOSI, SPI_MASTER_MISO, SPI_MASTER_SCK), NC, &capabilities);
+
+    /* SPI master. */
+    spi_init(&spi_obj, false, SPI_MASTER_MOSI, SPI_MASTER_MISO, SPI_MASTER_SCK, NC);
+
+    /* Skip last two configs (high frequency) since we may not be able to abort before the transmission ends. */
+    for (uint32_t tc_id = 0; tc_id < ((sizeof(test_cases) / sizeof(config_test_case_t)) - 2); tc_id++) {
+
+        if (capabilities.word_length & (1 << (test_cases[tc_id].symbol_size - 1))) {
+
+            spi_format(&spi_obj, test_cases[tc_id].symbol_size, test_cases[tc_id].mode, test_cases[tc_id].bit_ordering);
+            spi_frequency(&spi_obj, test_cases[tc_id].freq_hz);
+
+            if (test_cases[tc_id].symbol_size < 8) {
+                p_tx_buf = tx_buff_uint8;
+                p_rx_buf = rx_buff_uint8;
+                p_fill_sym = (void *)&fill_symbol_8;
+            } else if (test_cases[tc_id].symbol_size < 16) {
+                p_tx_buf = tx_buff_uint16;
+                p_rx_buf = rx_buff_uint16;
+                p_fill_sym = (void *)&fill_symbol_16;
+            } else {
+                p_tx_buf = tx_buff_uint32;
+                p_rx_buf = rx_buff_uint32;
+                p_fill_sym = (void *)&fill_symbol_32;
+            }
+
+            for (uint32_t mode_id = 0; mode_id < (sizeof(dma_modes) / sizeof(DMAUsage)); mode_id++) {
+                expected_spi_obj = &spi_obj;
+                expected_event.transfered = 0; // Abort will have to be indicated here
+                handler_called = false;
+
+                /* Ensure that transfer is aborted  is immediately after it is triggered. */
+                core_util_critical_section_enter();
+
+                const bool async_status = spi_transfer_async(&spi_obj, p_tx_buf, SYM_CNT, p_rx_buf, SYM_CNT, p_fill_sym, async_handler, (void *)&some_ctx, dma_modes[mode_id]);
+
+                TEST_ASSERT_EQUAL(true, async_status);
+
+                /* Transfer of 100 bytes @ 1 MHz should take about 100 us.
+                 * This should be enough time to call abort during the transfer.
+                 */
+                spi_transfer_async_abort(&spi_obj);
+
+                core_util_critical_section_exit();
+
+                wait_ms(2);
```

Comment:
Could you add a comment here and explain the reason for this `wait_ms(2)`?

Reply:
fixed.